### PR TITLE
doc: explain the maxGraph mission and status in the README

### PR DESCRIPTION
**Pull request by tbouffard**
**Summary**

The README is the main entry point of the lib. It should provide the main information about the project and should attract potential users.
Currently, as noticed by @mcyph in https://github.com/maxGraph/maxGraph/pull/85#issuecomment-1120377371,It is a mix of technical information and outdated sentences about the good old mxGraph.

This Pull Request is a first attempt to improve things. In short
- First present what the lib does and the project status.
- Then, provide a code example and a gif to let potential users see the rendering at a glance.
- Remove inaccurate content and refresh the project history.

I am sure that the layout and the style can be improved, but IMHO, it is a good start that we will able to improve iteratively later. 

**Description for the changelog**

doc: explain what maxGraph does an the project status in the README

**Other info**

Since I am not a native English speaker, I would be happy if someone could improve the wording 😸 .

This contribution is done under the terms of the Apache 2.0 license as stated in https://www.apache.org/licenses/LICENSE-2.0.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,97 +1,161 @@
...
-```
+`maxGraph` APIs are not fully compatible with mxGraph APIs. The concepts are the same, so experienced mxGraph users should be able to switch from mxGraph to maxGraph without issues.
 
-# Transition to Apache 2.0 license
+The main changes is the Internet Explorer support removal (including VML support).
 
-This project is published under the terms of a [Modified-Apache-2.0](LICENSE) license and currently preparing to move back to the standard Apache 2.0 license.  
-From now on, all contributions to this project must be under the terms of the Apaches 2.0 license as stated in https://www.apache.org/licenses/LICENSE-2.0.  
-See https://github.com/maxGraph/maxGraph/issues/89 for more details.
+### Migrating JavaScript
```
Comment (tbouffard) [inline README.md]: ❓ please provide suggestion for the JS and TS migration title. I'm out of ideas.

File: README.md [hunk 2]
```diff
@@ -1,97 +1,161 @@
...
+See https://github.com/maxGraph/maxGraph/issues/89 for more details.
+
+
+## Getting Started
+
+Assuming your page defines an element with the id `graph-container`, the following will display a rectangle connected to an orange circle.
+```typescript
+import {type CellStyle, Graph, InternalEvent} from '@maxgraph/core';
+
+const container = document.getElementById('graph-container') as HTMLElement;
```
Comment (tbouffard) [inline README.md]:
```suggestion
const container = <HTMLElement>document.getElementById('graph-container');
```

File: README.md [hunk 3]
```diff
@@ -1,97 +1,161 @@
...
+
+// Gets the default parent for inserting new cells. This
+// is normally the first child of the root (ie. layer 0).
+const parent = graph.getDefaultParent();
+graph.setPanning(true); // Use mouse right button for panning
+
+// Adds cells to the model in a single step
+graph.batchUpdate(() => {
+  const vertex01 = graph.insertVertex(parent, null, 'a regular rectangle', 10, 10, 100, 100);
+  const vertex02 = graph.insertVertex(parent, null, 'a regular ellipse', 350, 90, 50, 50, <CellStyle>{shape: 'ellipse', fillColor: 'orange'});
```
Comment (tbouffard) [inline README.md:63]: ℹ️  the method signature in the type is currently: `insertVertex: (...args: any[]) => Cell;`
So, cast to `CellStyle` to get code assist.

**Comment by mcyph** [comment]
A good start, would like to add something which states what mxGraph/maxGraph does too. Putting this together on the train but will revise this later:

> maxGraph is a TypeScript library which can display and allow interaction with vector diagrams. At a high level, it provides: 
> - **Nodes**, also known as **vertices** which are typically represented by shapes like rectangles.
> - **Edges** which can be lines and arrows which normally point between one node and another.
>
> It provides many of the diagramming features which would be expected by a piece of presentation software like Microsoft® PowerPoint™ or LibreOffice® Impress such as being able to resize, move or rotate nodes, but has a stronger focus on automatic layout algorithms and applications of [Graph Theory](https://en.wikipedia.org/wiki/Graph_theory). It is suited towards software which requires finer-grained customization of functionality than off-the-shelf packages.

Let me know if the above sounds ok to you, or if it could be improved. I think the wording could be made smoother.

**Comment by tbouffard** [comment]
> Let me know if the above sounds ok to you, or if it could be improved. I think the wording could be made smoother.

I like the idea to have a more detailed description of the lib and your proposal sounds good to me 💯 
I will read it again next week with fresh eyes and see if it has to be improved. Otherwise let's go for it!

File: README.md [hunk 4]
```diff
@@ -1,97 +1,167 @@
...
 
-In the project root directory, execute
+It provides many of the diagramming features which would be expected by a piece of presentation software like Microsoft® PowerPoint™ or LibreOffice® Impress such as being able to resize, move or rotate nodes, but has a stronger focus on automatic layout algorithms and applications of [Graph Theory](https://en.wikipedia.org/wiki/Graph_theory). It is suited towards software which requires finer-grained customization of functionality than off-the-shelf packages.
 
-```sh
-$ npm install
-```
 
-To watch the core package, execute
+The `maxGraph` library uses no third-party software, it requires no plugins and can be integrated in virtually any framework (it's vanilla JS).
```
Comment (tbouffard) [inline README.md:15]: This is kept from the original mxGraph README. I suggest we remove it as it doesn't provide valuable information.
```suggestion
```

**Comment by tbouffard** [comment]
@mcyph I have integrated in the README the proposal you made in https://github.com/maxGraph/maxGraph/pull/93#issuecomment-1152869294
I also did additional improvements in other paragraphs.

File: README.md (hunk 4 above)
Comment (tbouffard) [reply README.md:15]: keep for now

**Comment by tbouffard** [comment]
I am going to merge this PR: it is much more accurate than the current one and it can be improved later.

**Pull request merged**